core/drivers/stm32_bsec.c

```diff
@@ -713,8 +713,41 @@ static void bsec_dt_otp_nsec_access(void *fdt, int bsec_node)
 			}
 		}
 
-		if (!fdt_getprop(fdt, bsec_subnode, "st,non-secure-otp", NULL))
+		/* Handle different kinds of non-secure accesses */
+		if (fdt_getprop(fdt, bsec_subnode,
+				"st,non-secure-otp-provisioning", NULL)) {
+			bool locked = false;
+			bool lcked_next = false;
+
+			/* Check if write of OTP is locked */
+			if (stm32_bsec_read_permanent_lock(otp_id, &locked))
+				panic("BSEC: Couldn't read permanent lock at init");
+
+			/*
+			 * Check if fuses of the subnode
+			 * have the same lock status
+			 */
+			for (i = 1; i < (reg_size / sizeof(uint32_t)); i++) {
+				if (stm32_bsec_read_permanent_lock(otp_id + i,
+								   &lcked_next))
+					panic("Cannot read permanent lock");
+
+				if (locked != lcked_next) {
+					EMSG("Inconsistent status OTP id %u",
+					     otp_id + i);
+					locked = true;
+					continue;
```

Comment:
`continue` has no effect here even if the implementation looks correct to me. I suggest to remove it.

Reply:
Yes indeed...